src/Core/Grid/Definition/Factory/CountryGridDefinitionFactory.php

```diff
@@ -0,0 +1,254 @@
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollectionInterface;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\BulkActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ToggleColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\DataColumn;
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollectionInterface;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use PrestaShopBundle\Form\Admin\Type\YesAndNoChoiceType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+
+class CountryGridDefinitionFactory extends AbstractGridDefinitionFactory
+{
+    use BulkDeleteActionTrait;
+    use DeleteActionTrait;
+
+    public const GRID_ID = 'country';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getId(): string
+    {
+        return self::GRID_ID;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getName(): string
+    {
+        return $this->trans('Country', [], 'Admin.Global');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getColumns(): ColumnCollectionInterface
+    {
+        return (new ColumnCollection())
+            ->add(
+                (new BulkActionColumn('bulk'))
+                    ->setOptions([
+                        'bulk_field' => 'id_country',
+                    ])
+            )
+            ->add(
+                (new DataColumn('id_country'))
+                    ->setName($this->trans('ID', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'id_country',
+                    ])
+            )
+            ->add(
+                (new DataColumn('name'))
+                    ->setName($this->trans('Country', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'name',
+                    ])
+            )
+            ->add(
+                (new DataColumn('iso_code'))
+                    ->setName($this->trans('ISO Code', [], 'Admin.International.Feature'))
+                    ->setOptions([
+                        'field' => 'iso_code',
+                    ])
+            )
+            ->add(
+                (new DataColumn('call_prefix'))
+                    ->setName($this->trans('Call prefix', [], 'Admin.International.Feature'))
+                    ->setOptions([
+                        'field' => 'call_prefix',
+                    ])
+            )
+            ->add(
+                (new DataColumn('zone_name'))
+                    ->setName($this->trans('Zone', [], 'Admin.Global'))
+                    ->setOptions([
+                        'field' => 'zone_name',
+                    ])
+            )
+            //todo: change it to ToggleColumn when toggle status route is created
```

Comment:
If it is the only feature missing for the grid in this PR then I think it's worth handling it here. It shouldn't make the scope explode It's one command, one controller action, one JS extension to add, we have enough tools and example to do this one quickly. This way we only have one round of QA to do and the list page is fully functional.

Reply:
Intermediate PR so ok to do it later